azurelinuxagent/ga/monitor.py

```diff
@@ -172,90 +180,71 @@ def collect_event(self, evt_file_name):
+            event_list = TelemetryEventList()
+            event_dir = os.path.join(conf.get_lib_dir(), "events")
+            event_files = os.listdir(event_dir)
+            for event_file in event_files:
+                if not event_file.endswith(".tld"):
+                    continue
+                event_file_path = os.path.join(event_dir, event_file)
+                try:
+                    data_str = self.collect_event(event_file_path)
+                except EventError as e:
+                    logger.error("{0}", e)
+                    continue
+
+                try:
+                    event = parse_event(data_str)
+                    self.add_sysinfo(event)
+                    event_list.events.append(event)
+                except (ValueError, ProtocolError) as e:
+                    logger.warn("Failed to decode event file: {0}", e)
+                    continue
+
+            if len(event_list.events) == 0:
+                return
+
+            try:
+                protocol.report_event(event_list)
+            except ProtocolError as e:
+                logger.error("{0}", e)
+        except Exception as e:
+            logger.warn("Failed to send events: {0}", e)
+
+        return datetime.datetime.utcnow()
 
     def daemon(self):
-        period = datetime.timedelta(minutes=30)
-        protocol = self.protocol_util.get_protocol()        
-        last_heartbeat = datetime.datetime.utcnow() - period
 
-        # Create a new identifier on each restart and reset the counter
-        heartbeat_id = str(uuid.uuid4()).upper()
+        # Create a new identifier on each restart, reset the counter and all events
         counter = 0
-        while True:
-            if datetime.datetime.utcnow() >= (last_heartbeat + period):
-                last_heartbeat = datetime.datetime.utcnow()
-                incarnation = protocol.get_incarnation()
-                dropped_packets = self.osutil.get_firewall_dropped_packets(
-                                                    protocol.endpoint)
-
-                msg = "{0};{1};{2};{3}".format(
-                    incarnation, counter, heartbeat_id, dropped_packets)
-
-                add_event(
-                    name=AGENT_NAME,
-                    version=CURRENT_VERSION,
-                    op=WALAEventOperation.HeartBeat,
-                    is_success=True,
-                    message=msg,
-                    log_event=False)
-
-                counter += 1
-
-                io_errors = IOErrorCounter.get_and_reset()
-                hostplugin_errors = io_errors.get("hostplugin")
-                protocol_errors = io_errors.get("protocol")
-                other_errors = io_errors.get("other")
-
-                if hostplugin_errors > 0 \
-                        or protocol_errors > 0 \
-                        or other_errors > 0:
-
-                    msg = "hostplugin:{0};protocol:{1};other:{2}"\
-                        .format(hostplugin_errors,
```

Comment:
Did you just re-format the file?  It's hard to see real changes when the formatting gets adjusted this much.  Keep the format changes because I want them, but consider doing them in a distinct phase to make code review easier.

Reply:
I usually keep reformats/whitespace to separate commits, but in a review this size I understand it becomes too difficult to look at them separately.